### PR TITLE
feature - config linter, add github templates and add commitlint

### PR DESCRIPTION
**Pull request by kevfarid**
## Checklist ✅

- [x] I have followed the [contributor guideline](https://github.com/Platzi-Master-C8/gethired-commons/blob/main/CONTRIBUTING.md)
- [ ] Tests are included
- [x] Documentation is changed or added
- [x] Related issue has been created
- [x] Commits messages follows [commit guideline](https://github.com/Platzi-Master-C8/gethired-base/blob/main/CONTRIBUTING.md/#Commits)

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -1,25 +1,109 @@
...
+  root: true,
+  env: {
+    browser: true,
+    es2021: true,
+    node: true,
+    'jest/globals': true
+  },
+  extends: [
+    'plugin:react/recommended',
+    'plugin:import/typescript',
```
Comment (j0s3v4rg4s) [inline .eslintrc.js]: all configurations from Typescript should be in a typescript branch

File: .eslintrc.js [hunk 2]
```diff
@@ -1,25 +1,109 @@
...
+    'react/sort-prop-types': 'error',
+    'react/state-in-constructor': 'off',
+    'react/static-property-placement': 'off',
+    'react/react-in-jsx-scope': 'off',
+    'jsx-a11y/anchor-is-valid': 'off',
+    'react/no-arrow-function-lifecycle': 'off',
+    'react/no-invalid-html-attribute': 'off',
+    'react/no-unused-class-component-methods': 'off',
+    'jsx-a11y/alt-text': 'off',
+    'react/function-component-definition': 'off',
```
Comment (j0s3v4rg4s) [inline .eslintrc.js]: I'm not sure if is a good idea to take off all these rules, with airbnb rules I thinks that is enough

File: .prettierrc.json [hunk 3]
```diff
@@ -1,6 +1,14 @@
...
+  "trailingComma": "all",
+  "overrides": [
+    {
+      "files": "*.d.ts",
+      "options": {
+        "trailingComma": "es5"
+      }
+    }
+  ]
+}
```
Comment (j0s3v4rg4s) [inline .prettierrc.json]: this configuration could be a discussion form all team

File: package.json [hunk 4]
```diff
@@ -40,6 +40,19 @@
...
     "eslint-plugin-react": "^7.27.1",
     "eslint-plugin-react-hooks": "^4.3.0",
+    "@commitlint/cli": "^15.0.0",
+    "@commitlint/config-conventional": "^15.0.0",
+    "@testing-library/dom": "8.11.1",
+    "@testing-library/jest-dom": "5.15.0",
+    "@testing-library/react": "12.1.2",
+    "@types/react": "17.0.34",
+    "@typescript-eslint/eslint-plugin": "5.3.1",
+    "@typescript-eslint/parser": "5.3.1",
```
Comment (j0s3v4rg4s) [inline package.json]: same here, typescript configuration should be in a different branch

File: prettier.config.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+  overrides: [
+    {
+      files: '*.d.ts',
+      options: {
+        // This is needed for TypeScript 3.2 support
+        trailingComma: 'es5',
+      },
+    },
+  ],
+};
```
Comment (j0s3v4rg4s) [inline prettier.config.js]: already there is a `.prettierrc.json` so I think this file is not necessary

File: .eslintrc.js (hunk 1 above)
Comment (yadurani) [reply .eslintrc.js]: Please all configuration of typescript in other branch, by voting it was decided that the base template will be used with proptypes.

File: .prettierrc.json (hunk 3 above)
Comment (yadurani) [reply .prettierrc.json]: I will create a vote to review these points

File: src/App.jsx [hunk 6]
```diff
@@ -1,15 +1,15 @@
...
+import { ThemeProvider } from '@mui/material';
+import { THEME } from 'Constants/theme.constant';
+import './style.scss';
+
+const App = () => (
+	<React.StrictMode>
+		<ThemeProvider theme={THEME}>
+			<Theme />
+		</ThemeProvider>
+	</React.StrictMode>
```
Comment (j0s3v4rg4s) [inline src/App.jsx]: too much indentation space

**Review by j0s3v4rg4s — DISMISSED** [review]

**Review by yadurani — DISMISSED** [review]

File: yarn.lock [hunk 7]
```diff
@@ -6031,7 +6031,7 @@ semver@7.3.5, semver@^7.2.1, semver@^7.3.2, semver@^7.3.4, semver@^7.3.5:
 
 semver@^6.0.0, semver@^6.1.1, semver@^6.1.2, semver@^6.3.0:
   version "6.3.0"
-  resolved "https://registry.yarnpkg.com/semver/-/semver-6.3.0.tgz#ee0a64c8af5e8ceea67687b133761e1becbd1d3d"
+  resolved "https://registry.npmjs.org/semver/-/semver-6.3.0.tgz"
```
Comment (yadurani) [inline yarn.lock:6034]: Please do not version this file, because we are going to start to have conflicts because some of us use npm and others

**Review by j0s3v4rg4s — DISMISSED** [review]

**Review by j0s3v4rg4s — APPROVED** [review]

**Review by yadurani — APPROVED** [review]

**Pull request merged**